### ISSUE TITLE
house number

### ISSUE THREAD

**Issue by lgualtieri75** [ocpi/ocpi#278]
The other protocols have a distinct field for "house number" in the address.

**Comment by poohsen** [comment]
That's an observation. 
Do you have a proposal and some usecases to substantiate it or did you just want to spread awareness?

**Comment by lgualtieri75** [comment]
Of course, it's an observation and I don't know if there is a proper way to make a proposal.
The point is that when you have to deal with different protocols, especially the federation protocols you have to face this problem. Of course as CPO you can push address and house num in to the same field, but the EMSP has to deal with the mixed cases, and splitting road name and house numbers is not so straightforward.
For example the upcoming OICP 2.3 from Hubject is going to make it mandatory.

https://github.com/hubject/oicp/blob/master/OICP-2.3/Realease_Notes.asciidoc#OICP2.3DataTypeChanges

thanks

**Comment by simonschllng** [comment]
I would vote to remove the house number field. I'm sorry for not mentioning that many years ago.

**Comment by ahzf** [comment]
@simonschllng You hate computers, or ;)? In IT is always the better way to make something explicit because as someone from the other side of the planet it is easy to combine streetname + " " + housenumber, but it might be very very difficult so split a combined string into streetname + housenumber without proper local knowledge.

**Comment by RobertdeLeeuw-IHomer** [comment]
> I would vote to remove the house number field. I'm sorry for not mentioning that many years ago.

OCPI current has no house number field, so we cannot remove it.

**Comment by jeromegamez** [comment]
If I'm not mistaken, there are addresses without house numbers, and not all house numbers are actual numbers. Out of curiosity, what's the benefit of having a house number as a distinct field and, as an extension, making it required?

**Comment by poohsen** [comment]
you're right @jeromegamez . I'm surprised Hubject wants to do that. It will be difficult to implement in Ireland and the UK AFAIK.

**Comment by webprofusion-chrisc** [comment]
As an aside, Public EV charging equipment in general does *not* have an actual postal/street address, just an approximate address where the equipment might be found that's close to the site latitude/longitude. Most of it is in car parks.

Equipment addresses are generally present as a navigational aid but for the most part they should never be considered factual, only approximate guidance. 

Private charging may have a street building numbers, or charging operated by businesses/local governments but the postal address is only relevant for postal communications or billing relationships, not for users of charging location (drivers).

It's common in different countries for houses/buildings to have a name instead of a number but their data field is a string anyway. New buildings often have no allocated postal address and are referred to by their planning Lot number.

So the main issue here seems to be how to transform data to/from services with (some) fine grained address components like Hubject but realistically the lowest common denominator address storage is AddressLine1, AddressLine2, Town/City, Region, Postal Code, Country and in reality systems may even combine address strings into even fewer fields.

If you do need to transform between address formats you would likely need to use either commercial address transformation services or reverse geocoding of latitude/longtitude to the nearest address, or parse it yourself using an attempt at country specific rules.
